### PR TITLE
Feat: Implement interaction modal for Polls

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
This implements the interaction modal for Polls, per #32445.

Screenshot 2024-10-21 at 17 48 30

The approach I've taken here may not be _the_ best with tunneling the `status` object down to the poll, however, current the Poll component is connected, which handles the voting action itself, where as elsewhere we're already have the status as a prop (i.e., StatusActionBar)

I've tried to go for the most minimal changes required to ship this as a feature.

### CODE REVIEW

**Comment by ThisIsMissEm** [comment]
n.b., I'm not totally sold on the interaction's description, but that's something we can iterate on.

**Comment by samhenrigold** [comment]
Just to give you a heads up, we’re planning to rearrange this modal to make it easier for people to get at a glance. This is a useful addition, though, and we’ll let you know when those designs are ready to go so updates (if any) can be made 👍

**Comment by ThisIsMissEm** [comment]
@samhenrigold nice, but could this land before the redesign?

also, you could do "home server or username", and then do an account search to allow them to pick which server to go to based on accounts with their username/acct

**Comment by samhenrigold** [comment]
> could this land before the redesign

I have no idea how long the redesign is gonna take, it depends largely on how much other work (I have three other clients in addition to Mastodon) I have to cover. I would like to have at least a design done this week or next before I have to bunker down and work on quote posts for the foreseeable future. 

> also, you could do "home server or username", and then do an account search to allow them to pick which server to go to based on accounts with their username/acct

like an auto-complete pattern?

**Comment by ThisIsMissEm** [comment]
> > could this land before the redesign
> 
> I have no idea how long the redesign is gonna take, it depends largely on how much other work (I have three other clients in addition to Mastodon) I have to cover. I would like to have at least a design done this week or next before I have to bunker down and work on quote posts for the foreseeable future.

Yeah, so I'm not sure why we'd block a first pass at implementing this on completely redesigning the modal — we still need to trigger it & have it do something with the right contextual information 

> > also, you could do "home server or username", and then do an account search to allow them to pick which server to go to based on accounts with their username/acct
> 
> like an auto-complete pattern?

Yeah, sometimes like that. Current we auto-complete from known domains, if you start with @ then we could auto complete from known user accounts?

**Comment by renchap** [comment]
Let's merge this without the redesign, it can be done at a later step.

Can you remove the "and have your say" part?

File: app/javascript/mastodon/features/interaction_modal/index.jsx [hunk 1]
```diff
@@ -377,6 +378,11 @@ class InteractionModal extends React.PureComponent {
       title = <FormattedMessage id='interaction_modal.title.follow' defaultMessage='Follow {name}' values={{ name }} />;
       actionDescription = <FormattedMessage id='interaction_modal.description.follow' defaultMessage='With an account on Mastodon, you can follow {name} to receive their posts in your home feed.' values={{ name }} />;
       break;
+    case 'vote':
+      icon = <Icon id='tasks' icon={InsertChartIcon} />;
+      title = <FormattedMessage id='interaction_modal.title.vote' defaultMessage="Vote in {name}'s poll" values={{ name }} />;
+      actionDescription = <FormattedMessage id='interaction_modal.description.vote' defaultMessage='With an account on Mastodon, you can vote in this poll and have your say.' />;
```
Comment (ThisIsMissEm) [inline app/javascript/mastodon/features/interaction_modal/index.jsx:384]:
```suggestion
      actionDescription = <FormattedMessage id='interaction_modal.description.vote' defaultMessage='With an account on Mastodon, you can vote in this poll.' />;
```

File: app/javascript/mastodon/locales/en.json [hunk 2]
```diff
@@ -385,6 +385,7 @@
   "interaction_modal.description.follow": "With an account on Mastodon, you can follow {name} to receive their posts in your home feed.",
   "interaction_modal.description.reblog": "With an account on Mastodon, you can boost this post to share it with your own followers.",
   "interaction_modal.description.reply": "With an account on Mastodon, you can respond to this post.",
+  "interaction_modal.description.vote": "With an account on Mastodon, you can vote in this poll and have your say.",
```
Comment (ThisIsMissEm) [inline app/javascript/mastodon/locales/en.json:388]:
```suggestion
  "interaction_modal.description.vote": "With an account on Mastodon, you can vote in this poll.",
```

**Review by ThisIsMissEm — COMMENTED** [review]
Renchaps Feedback

**Comment by ThisIsMissEm** [comment]
@renchap done! (If need be, can the team update the branch? I only have my phone whilst I'm in hospital)

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**